Islandora/islandora
# Add core image server settings

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2469

Related PRs:
* https://github.com/Islandora/islandora_paged_content/pull/158
* https://github.com/Islandora/islandora_internet_archive_bookreader/pull/136
* https://github.com/Islandora/islandora_openseadragon/pull/102

# What does this Pull Request do?

Creates a new configuration space in core for image server settings. This will be shared between Paged Content, the IABV and Openseadragon.

# What's new?

An admin form, tokens and some helper functions.

# How should this be tested?

Try setting as type as none, Djatoka and IIIF, ensure the form saves your settings. Set using different tokens for IIIF, etc. If you have a custom setup for your site, try it here and ensure it is still supported.

# Additional Notes:

Example:
* Does this change the interface, add a new feature, or otherwise change behaviours that would require updating documentation? Yes, documentation changes could be needed in the wiki. I have altered the old configuration forms to direct people here to configure their image server.
* Does this change add any new dependencies? no
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? It will if settings are not auto-configured until someone fixes them.

# Interested parties
@Islandora/7-x-1-x-committers @jonathangreen
